tests/test_closespider.py

```diff
@@ -34,6 +34,19 @@ def test_closespider_pagecount(self):
         pagecount = crawler.stats.get_value("response_received_count")
         self.assertTrue(pagecount >= close_on)
 
+    @defer.inlineCallbacks
+    def test_closespider_pagecount_no_item(self):
+        close_on = 5
+        crawler = get_crawler(
+            FollowAllSpider,
+            {"CLOSESPIDER_PAGECOUNT_NO_ITEM": close_on},
+        )
+        yield crawler.crawl(mockserver=self.mockserver)
+        reason = crawler.spider.meta["close_reason"]
+        self.assertEqual(reason, "closespider_pagecount_no_item")
+        pagecount = crawler.stats.get_value("response_received_count")
+        self.assertTrue(pagecount >= close_on)
```

Comment:
This does not test what makes this feature different from `CLOSESPIDER_PAGECOUNT`, though. Maybe you could implement an alternative to `FollowAllSpider` that allows testing that difference, i.e. build a test that would fail with `CLOSESPIDER_PAGECOUNT`.